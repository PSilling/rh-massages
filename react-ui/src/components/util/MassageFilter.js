// react imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// util imports
import Util from '../../util/Util';
import _t from '../../util/Translations';

/**
 * Custom Masssage filtering component (search input with status checkbox).
 */
class MassageFilter extends Component {

  render() {
    return (
      <div className="row">
        <div className="col-md-5">
          <div className="input-group" style={{ 'marginBottom': '15px' }}>
              <span className="input-group-addon"><span className="glyphicon glyphicon-search"></span></span>
              <input type="search" value={this.props.value} className="form-control"
                placeholder={ _t.translate('Search') } onChange={this.props.onSearchChange}
                autoFocus onFocus={Util.moveCursorToEnd}
              />
          </div>
        </div>

        <div className="form-group col-md-5">
          <div className="col-md-5 col-md-offset-1">
            <input id="fromInput" value={this.props.from}
              onChange={this.props.onFromChange} className="form-control"
              type="date" placeholder={ _t.translate('From') }
            />
          </div>

          <div className="col-md-1" style={{ 'marginTop': '6px' }}>
            <label htmlFor="toInput">{ "â€“" }</label>
          </div>

          <div className="col-md-5">
            <input id="toInput" value={this.props.to}
              onChange={this.props.onToChange} className="form-control"
              type="date" placeholder={ _t.translate('To') }
            />
          </div>
        </div>

        <div className="form-group col-md-2" style={{ 'marginTop': '6px' }}>
          <label className="checkbox-inline">
            <input type="checkbox" onChange={this.props.onCheck} checked={this.props.checked} />
            <strong>{ _t.translate('only free massages') }</strong>
          </label>
        </div>
      </div>
    );
  }
}

MassageFilter.propTypes = {
  value: PropTypes.string, // current value of the search input
  from: PropTypes.string, // current value of the from date input
  to: PropTypes.string, // current value of the to date input
  checked: PropTypes.bool, // whether the free Massages only checkbox should be checked
  onCheck: PropTypes.func, // function called on free Massages only checkbox change
  onFromChange: PropTypes.func, // function called on from date value change
  onToChange: PropTypes.func, // function called on to date value change
  onSearchChange: PropTypes.func // function called on search value change
};

export default MassageFilter
