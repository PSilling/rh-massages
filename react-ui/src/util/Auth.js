// Auth handler using Keycloak
import Keycloak from 'keycloak-js';

// util imports
import Util from './Util';

/**
 * Keycloak authorization holder. Functions called on Auth work with the Keycloak
 * conncetion variable.
 */
var Auth = function() {};

Auth.keycloak = Keycloak();

/**
 * Register new and update old Clients after Keycloak authorization success.
 */
Auth.keycloak.onAuthSuccess = () => {
  Util.get(Util.CLIENTS_URL + 'my/subscribed', (json) => {
    Auth.subscribed = json;
    Util.put(Util.CLIENTS_URL, Auth.getClient(), () => {
    }, false);
  });
}

Auth.subscribed = true;

Auth.isAuthenticated = () => {
  return Auth.keycloak.authenticated;
}

Auth.isAdmin = () => {
  return Auth.keycloak.hasRealmRole("admin");
}

Auth.getToken = () => {
  return Auth.keycloak.token;
}

Auth.getSub = () => {
  return Auth.keycloak.subject;
}

/**
 * Generates a new Client based on server Client representation.
 */
Auth.getClient = () => {
  return {
    sub: Auth.keycloak.subject, email: Auth.keycloak.idTokenParsed.email,
    name: Auth.keycloak.idTokenParsed.given_name, surname: Auth.keycloak.idTokenParsed.family_name,
    subscribed: Auth.subscribed
  };
}

export default Auth
