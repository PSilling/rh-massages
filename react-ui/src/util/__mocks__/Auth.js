const isAdmin = jest.fn().mockReturnValue(true);
const isAuthenticated = jest.fn().mockReturnValue(true);
const subscribed = true;
const getToken = jest.fn().mockReturnValue("secret");
const getSub = jest.fn().mockReturnValue("test");
const getClient = jest.fn().mockReturnValue({
  sub: "test", email: "test@email.com",
  name: "Test", surname: "User",
  subscribed: true
});

module.exports = {
  isAdmin,
  isAuthenticated,
  subscribed,
  getToken,
  getSub,
  getClient
}
