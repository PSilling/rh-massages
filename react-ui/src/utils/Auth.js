// Auth "service"
const Auth = {

  isAuthenticated() {
    return true;
  },

  isAdmin() {
    return true;
  }
}

export default Auth
