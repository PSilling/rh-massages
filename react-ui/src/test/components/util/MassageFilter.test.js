// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import MassageFilter from '../../../components/util/MassageFilter';
import Auth from '../../../util/Auth';

// test mocks
jest.mock('../../../util/Auth');

afterEach(() => {
  jest.resetAllMocks();
});

test('renders admin content with correct props', () => {
  const testSelectFunction = jest.fn(),
        testFreeFunction = jest.fn(),
        testFilterFunction = jest.fn(),
        testRenderer = TestRenderer.create(
          <MassageFilter
            filter="test"
            select={false}
            free={true}
            onSelectCheck={testSelectFunction}
            onFreeCheck={testFreeFunction}
            onFilterChange={testFilterFunction}
          />
        ),
        testInstance = testRenderer.root;

  let inputs = testInstance.findAllByType('input'),
      treeJSON = testRenderer.toJSON();

  expect(inputs[0].props.value).toBe("test");
  expect(inputs[0].props.onChange).toBe(testFilterFunction);
  expect(inputs[1].props.onChange).toBe(testFreeFunction);
  expect(inputs[1].props.checked).toBe(true);
  expect(inputs[2].props.onChange).toBe(testSelectFunction);
  expect(inputs[2].props.checked).toBe(false);
  expect(treeJSON).toMatchSnapshot();
});

test('hides non-admin inputs correctly', () => {
  Auth.isAuthenticated = jest.fn().mockImplementation(() => true);
  Auth.isAdmin = jest.fn().mockImplementation(() => false);

  const testFunction = jest.fn(),
        testRenderer = TestRenderer.create(
          <MassageFilter free={false} onFreeCheck={testFunction} />
        ),
        testInstance = testRenderer.root;

  let inputs = testInstance.findAllByType('input');

  expect(inputs.length).toBe(2);
  expect(inputs[1].props.onChange).toBe(testFunction);
  expect(inputs[1].props.checked).toBe(false);
});
