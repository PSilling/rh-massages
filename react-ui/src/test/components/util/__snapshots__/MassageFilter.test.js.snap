// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders admin content with correct props 1`] = `
<div
  className="row"
>
  <div
    className="col-md-8"
  >
    <div
      className="input-group"
      style={
        Object {
          "marginBottom": "15px",
        }
      }
    >
      <span
        className="input-group-addon"
      >
        <span
          className="glyphicon glyphicon-search"
        />
      </span>
      <input
        autoFocus={true}
        className="form-control"
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Filtrování"
        type="search"
        value="test"
      />
    </div>
  </div>
  <div>
    <div
      className="form-group col-md-4 text-right"
      style={
        Object {
          "marginTop": "6px",
        }
      }
    >
      <label
        className="checkbox-inline"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <strong>
          pouze volné masáže
        </strong>
      </label>
      <label
        className="checkbox-inline"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <strong>
          vybírat události
        </strong>
      </label>
    </div>
  </div>
</div>
`;
